src/hooks/useCreditCardInput.tsx

```diff
@@ -91,18 +95,21 @@ function useCreditCardInput() {
         }
 
         props.onChange && props.onChange(e);
-
-        const cardNumberError = getCardNumberError(cardNumber);
+        let cardNumberError;
+        typeof cardErrorMessage !== 'undefined'
```

Comment:
How about moving this logic to the `getCardNumberError` function definition, with default parameters?.